lib/merit/excess.rb

```diff
@@ -0,0 +1,38 @@
+module Merit
+  class Excess
+    def initialize(order)
+      @order = order
+    end
+
+    # Public: number_of_events
+    #
+    # Given a duration in hours, this method determines the amount of times
+    # production exceeds consumption by the specified duration.
+    #
+    def number_of_events(duration)
+      possible_excess_events.inject(0) do |result, (excess, streak)|
+        result + (excess ? (streak.size / duration) : 0)
+      end
+    end
+
+    private
+
+    def possible_excess_events
```

Comment:
Here's an interesting thing... if you filter out the "non-excess" chunks and then cache the result, a single call to `number_of_events` should run in about the same time, but *multiple* calls are quite a bit faster (e.g. if you want to use duration=1, duration=2, duration=4, etc):

```ruby
def possible_excess_events
  @events ||= net_load.values.chunk { |point| point > 0 }.select(&:first)
end
```

... then the `number_of_events` inject block can become:

```ruby
possible_excess_events.inject(0) do |result, (_, streak)|
  result + (streak.size / duration)
end
```

I used LONG_EVENTS and SHORT_EVENTS from the benchmark and then ran `number_of_events` with durations 1, 2, 4, and 8, and got the following results:

| Method | LONG_EVENTS | SHORT_EVENTS |
| ---- | ----: | ----: |
| Non-cached / filtered | 9ms | 19ms |
| Cached / filtered | 2.2ms | 4ms |

That's should be quite an improvement if the merit curves have short events.

--

On a sidenote, the method names are a bit unwieldy. What about just `events`, `production`, and `consumption`?